components/renku_data_services/namespace/orm.py

```diff
@@ -74,21 +74,40 @@ class NamespaceORM(BaseORM):
     )
     user: Mapped[UserORM | None] = relationship(lazy="joined", init=False, repr=False, viewonly=True)
 
+    @property
+    def created_by(self) -> str:
+        """User that created this namespace."""
+        if self.group is not None:
+            return self.group.created_by
+        elif self.user_id:
+            return self.user_id
+        raise errors.ProgrammingError(
+            message=f"Found a namespace {self.slug} that has no group or user associated with it."
+        )
+
+    @property
+    def creation_date(self) -> datetime | None:
+        """When this namespace was created."""
+        return self.group.creation_date if self.group else None
+
     def dump(self) -> models.Namespace:
```

Comment:
It seems that the `dump()` method could avoid branching by using some of the new `properties`. Maybe a new one needs to be added for `self.underlying_resource_id`.